service/matching/handler/handler.go

```diff
@@ -415,6 +415,56 @@ func (h *handlerImpl) GetTaskListsByDomain(
 	return response, hCtx.handleErr(err)
 }
 
+func (h *handlerImpl) UpdateTaskListPartitionConfig(
+	ctx context.Context,
+	request *types.MatchingUpdateTaskListPartitionConfigRequest,
+) (resp *types.MatchingUpdateTaskListPartitionConfigResponse, retError error) {
+	defer func() { log.CapturePanic(recover(), h.logger, &retError) }()
+
+	domainName := h.domainName(request.DomainUUID)
+	hCtx := h.newHandlerContext(
+		ctx,
+		domainName,
+		request.TaskList,
+		metrics.MatchingUpdateTaskListPartitionConfigScope,
+	)
+
+	sw := hCtx.startProfiling(&h.startWG)
+	defer sw.Stop()
+
+	if ok := h.userRateLimiter.Allow(quotas.Info{Domain: domainName}); !ok {
+		return nil, hCtx.handleErr(errMatchingHostThrottle)
+	}
+
+	response, err := h.engine.UpdateTaskListPartitionConfig(hCtx, request)
+	return response, hCtx.handleErr(err)
+}
+
+func (h *handlerImpl) RefreshTaskListPartitionConfig(
+	ctx context.Context,
+	request *types.MatchingRefreshTaskListPartitionConfigRequest,
+) (resp *types.MatchingRefreshTaskListPartitionConfigResponse, retError error) {
+	defer func() { log.CapturePanic(recover(), h.logger, &retError) }()
+
+	domainName := h.domainName(request.DomainUUID)
+	hCtx := h.newHandlerContext(
+		ctx,
+		domainName,
+		request.TaskList,
+		metrics.MatchingRefreshTaskListPartitionConfigScope,
+	)
+
+	sw := hCtx.startProfiling(&h.startWG)
+	defer sw.Stop()
+
+	if ok := h.userRateLimiter.Allow(quotas.Info{Domain: domainName}); !ok {
```

Comment:
Refresh will be an internal call and shouldn't consume from user rate limit budget. WDYT?

Reply:
yeah, that makes sense